examples/arduino-uno/src/bin/uno-timer.rs

```diff
@@ -0,0 +1,136 @@
+#![no_std]
+#![no_main]
+#![feature(abi_avr_interrupt)]
+
+/*
+ This is derived from Rahix' comment to
+https://github.com/Rahix/avr-hal/issues/75
+and then modernized to account for API drift since 2020
+
+*/
+
+use arduino_hal::hal::port::PB5;
+use arduino_hal::port::mode::Output;
+use arduino_hal::port::Pin;
+use arduino_hal::prelude::*;
+use avr_device::atmega328p::tc1::tccr1b::CS1_A;
+use avr_device::atmega328p::TC1;
+use core::mem;
+use panic_halt as _;
+use ufmt::{uWrite, uwriteln};
+
+struct InterruptState<LED> {
+    blinker: Pin<Output, LED>,
+}
+
+static mut INTERRUPT_STATE: mem::MaybeUninit<InterruptState<PB5>> = mem::MaybeUninit::uninit();
+
+#[arduino_hal::entry]
+fn main() -> ! {
+    let dp = arduino_hal::Peripherals::take().unwrap();
+    let pins = arduino_hal::pins!(dp);
+
+    let mut serial = arduino_hal::default_serial!(dp, pins, 57600);
+    ufmt::uwriteln!(&mut serial, "Hello from Arduino!\r").void_unwrap();
+
+    let led = pins.d13.into_output();
+
+    unsafe {
+        // SAFETY: Interrupts are not enabled at this point so we can safely write the global
+        // variable here.  A memory barrier afterwards ensures the compiler won't reorder this
+        // after any operation that enables interrupts.
+        INTERRUPT_STATE = mem::MaybeUninit::new(InterruptState { blinker: led });
+        core::sync::atomic::compiler_fence(core::sync::atomic::Ordering::SeqCst);
+    }
+
+    //
+
+    let tmr1: TC1 = dp.TC1;
+
+    rig_timer(&tmr1, &mut serial);
+
+    // Enable interrupts globally, not a replacement for the specific interrupt enable
+    unsafe {
+        // SAFETY: Not inside a critical section and any non-atomic operations have been completed
+        // at this point.
+        avr_device::interrupt::enable();
+    }
+
+    if true {
+        ufmt::uwriteln!(
+            &mut serial,
+            "configured timer output compare register = {}",
+            tmr1.ocr1a.read().bits()
+        )
+        .void_unwrap();
+    }
+    loop {
+        if true {
+            avr_device::asm::sleep()
+        } else {
+            arduino_hal::delay_ms(100);
+
+            ufmt::uwriteln!(&mut serial, "counter = {}", tmr1.tcnt1.read().bits()).void_unwrap();
+        }
+    }
+}
+
+pub const fn calc_overflow(clock_hz: u32, target_hz: u32, prescale: u32) -> u32 {
+    /*
+    https://github.com/Rahix/avr-hal/issues/75
+    reversing the formula F = 16 MHz / (256 * (1 + 15624)) = 4 Hz
+     */
+    clock_hz / target_hz / prescale - 1
+}
+
+pub fn rig_timer<W: uWrite<Error = void::Void>>(tmr1: &TC1, serial: &mut W) {
+    /*
+     https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-7810-Automotive-Microcontrollers-ATmega328P_Datasheet.pdf
+     section 15.11
+    */
+
+    const ARDUINO_UNO_CLOCK_FREQUENCY_HZ: u32 = 16_000_000;
```

Comment:
Please reference the clock via the clock-type from `arduino-hal`: [`arduino_hal::DefaultClock`](https://rahix.github.io/avr-hal/arduino_hal/type.DefaultClock.html)

You can get the frequency using `arduino_hal::DefaultClock::FREQ`.